### PR TITLE
[INLONG-9957][Audit] Audit-service add local cache for openapi

### LINKED ISSUE #9957: [Feature][Audit] Audit-service add local cache for openapi
### Description
image

Audit-service add local cache for openapi to improve the concurrency of service

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by doleyzi**
### Prepare a Pull Request

- Fixes #9957 

### Motivation
image

*Audit-service add local cache for openapi to improve the concurrency of service*

### Modifications
*Synchronize mysql data to the local cache in real time, and the cache will provide services to the outside world.*

### CODE REVIEW

**Review by justinwwhuang — APPROVED** [review]

File: inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/utils/CacheUtils.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    public static String buildCacheKey(String logTs, String inlongGroupId, String inlongStreamId,
+            String auditId, String auditTag) {
+        StringBuilder keyBuilder = new StringBuilder();
+        return keyBuilder
+                .append(logTs)
+                .append(inlongGroupId)
+                .append(inlongStreamId)
+                .append(auditId)
+                .append(auditTag)
+                .toString();
```
Comment (aloyszhang) [inline inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/utils/CacheUtils.java:34]:
```suggestion
        return new StringBuilder()
                .append(logTs)
                .append(inlongGroupId)
                .append(inlongStreamId)
                .append(auditId)
                .append(auditTag)
                .toString();
```

File: inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/config/ConfigConstants.java [hunk 2]
```diff
@@ -123,4 +123,12 @@ public class ConfigConstants {
 
     public static final int MAX_INIT_COUNT = 2;
     public static final int RANDOM_BOUND = 10;
+
+    // Cache config
+    public static final String KEY_API_CACHE_MAX_SIZE = "api.cache.size";
```
Comment (aloyszhang) [inline inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/config/ConfigConstants.java:128]:
```suggestion
    public static final String KEY_API_CACHE_MAX_SIZE = "api.cache.max.size";
```

File: inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/sink/CacheSink.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        try {
+            StatData data = dataQueue.pull(pullTimeOut, TimeUnit.MILLISECONDS);
+            while (data != null) {
+                String cacheKey = CacheUtils.buildCacheKey(data.getLogTs(), data.getInlongGroupId(),
+                        data.getInlongStreamId(), data.getAuditId(), data.getAuditTag());
+                cache.put(cacheKey, data);
+                data = dataQueue.pull(pullTimeOut, TimeUnit.MILLISECONDS);
+            }
+        } catch (Exception e) {
+            LOG.error("Process exception! {}", e.getMessage());
```
Comment (aloyszhang) [inline inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/sink/CacheSink.java:80]: is it better to print the stack trace of exception?

File: inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/cache/CacheOfDay.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                    StatData data = new StatData();
+                    data.setLogTs(startTime);
+                    data.setInlongGroupId(inlongGroupID);
+                    data.setInlongStreamId(InlongStreamID);
+                    data.setAuditId(AuditId);
+                    data.setAuditTag(AuditTag);
+                    data.setCount(count);
+                    data.setSize(size);
+                    data.setDelay(delay);
+                    result.add(data);
```
Comment (aloyszhang) [inline inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/cache/CacheOfDay.java:132]:
```suggestion
                    StatData data = new StatData();
                    data.setLogTs(startTime);
                    data.setInlongGroupId(resultSet.getString(1));
                    data.setInlongStreamId(resultSet.getString(2));
                    data.setAuditId(resultSet.getString(3));
                    data.setAuditTag(resultSet.getString(4));
                    data.setCount(resultSet.getLong(5));
                    data.setSize(resultSet.getLong(6));
                    data.setDelay( resultSet.getLong(7));
                    result.add(data);
```

**Comment by aloyszhang** [comment]
There are caches for various time units like `CacheOfDay`,` CacheOfHour`, `CacheOfMinute10`, and `CacheOfMinute30`.
How about changing the class name to `DayCache`, `HourCache`, `TenMinutesCache`, and `HalfHourCache`?

**Comment by doleyzi** [comment]
> There are caches for various time units like `CacheOfDay`,` CacheOfHour`, `CacheOfMinute10`, and `CacheOfMinute30`. How about changing the class name to `DayCache`, `HourCache`, `TenMinutesCache`, and `HalfHourCache`?

This is a good suggestion and has been updated

**Review by gosonzhang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]
LGTM

**Pull request merged**